homeassistant/components/vicare/entity.py

```diff
@@ -60,3 +59,11 @@ def __init__(
                     DOMAIN,
                     f"{gateway_serial}_zigbee_{zigbee_ieee}",
                 )
+            elif len(parts) == 2:
+                ieee = device_serial.replace("zigbee-", "")
+                if len(ieee) == 16:
+                    self._attr_device_info["serial_number"] = "-".join(
+                        ieee.upper()[i : i + 2] for i in range(0, 16, 2)
+                    )
```

Comment:
```suggestion
            elif len(parts) == 2 and len((ieee := device_serial.removeprefix("zigbee-"))) == 16:
			    self._attr_device_info["serial_number"] = "-".join(
			        ieee.upper()[i:i + 2] for i in range(0, 16, 2)
			    )
```
It's a bit nitpicky, so feel free to treat it as such. But you can reduce the if-statement and use the walrus operator to set the ieee. Trade-off is that it might be less readable...